File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/library/genres/items/GenreItemsView.java
package com.github.anrimian.musicplayer.ui.library.genres.items;

import com.github.anrimian.musicplayer.domain.models.genres.Genre;
import com.github.anrimian.musicplayer.ui.library.common.compositions.BaseLibraryCompositionsView;
import com.github.anrimian.musicplayer.ui.utils.moxy.SingleStateByTagStrategy;

import moxy.viewstate.strategy.StateStrategyType;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;
import moxy.viewstate.strategy.alias.Skip;

public interface GenreItemsView extends BaseLibraryCompositionsView {

    String RENAME_STATE = "rename_state";

    @AddToEndSingle
    void showGenreInfo(Genre genre);

    @OneExecution
    void closeScreen();

    @Skip
    void showRenameGenreDialog(Genre genre);

    @StateStrategyType(value = SingleStateByTagStrategy.class, tag = RENAME_STATE)
    void showRenameProgress();

    @StateStrategyType(value = SingleStateByTagStrategy.class, tag = RENAME_STATE)
    void hideRenameProgress();
}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/library/genres/items/GenreItemsView.kt
package com.github.anrimian.musicplayer.ui.library.genres.items

import com.github.anrimian.musicplayer.domain.models.composition.Composition
import com.github.anrimian.musicplayer.domain.models.genres.Genre
import com.github.anrimian.musicplayer.ui.library.common.compositions.BaseLibraryCompositionsView
import moxy.viewstate.strategy.alias.AddToEndSingle
import moxy.viewstate.strategy.alias.OneExecution
import moxy.viewstate.strategy.alias.Skip

interface GenreItemsView : BaseLibraryCompositionsView<Composition> {

    @AddToEndSingle
    fun showGenreInfo(genre: Genre)

    @OneExecution
    fun closeScreen()

    @Skip
    fun showRenameGenreDialog(genre: Genre)

}